portal-api/src/v1/controllers/deal-status.controller.js

```diff
@@ -1,158 +1,142 @@
+const { HttpStatusCode } = require('axios');
 const { findOneDeal, updateDeal } = require('./deal.controller');
-
 const { addComment } = require('./deal-comments.controller');
-
 const { userHasAccessTo } = require('../users/checks');
-
 const validateStateChange = require('../validation/deal-status');
-
 const userCanSubmitDeal = require('./deal-status/user-can-submit-deal');
 const updateStatus = require('./deal-status/update-status');
 const createSubmissionDate = require('./deal-status/create-submission-date');
 const createMiaSubmissionDate = require('./deal-status/create-mia-submission-date');
-
 const updateSubmissionCount = require('./deal-status/update-submission-count');
-
 const sendStatusUpdateEmails = require('./deal-status/send-status-update-emails');
 const createApprovalDate = require('./deal-status/create-approval-date');
-
 const updateFacilityCoverStartDates = require('./deal-status/update-facility-cover-start-dates');
 const updateIssuedFacilities = require('./deal-status/update-issued-facilities');
 const updateSubmittedIssuedFacilities = require('./deal-status/update-submitted-issued-facilities');
 const createUkefIds = require('./deal-status/create-ukef-ids');
 const api = require('../api');
-
 const CONSTANTS = require('../../constants');
 
-exports.findOne = (req, res) => {
-  findOneDeal(req.params.id, (deal) => {
-    if (!deal) {
-      res.status(404).send();
-    } else if (!userHasAccessTo(req.user, deal)) {
-      res.status(401).send();
-    } else {
-      res.status(200).send(deal.status);
-    }
-  });
+/**
+ * Finds a specific deal by its ID and sends the deal's status as the response.
+ * @param {Request} req - The request object containing the parameters.
+ * @param {Response} res - The response object used to send the response.
+ * @returns {Response} - Response with status code, if Ok then with deal status
+ */
+exports.findOne = async (req, res) => {
```

Comment:
I know this is just you improving old code but this should probably be wrapped in a try/catch (and return a 500 error if catching an error)